tools/device.c

```diff
@@ -81,16 +82,21 @@ static bool bl_device__match(const char *sysname)
 		printf("No matched device found.\n");
 		goto enum_out;
 	}
+
 	dev = udev_device_new_from_syspath(udev, path);
+	if (!dev) {
+		fprintf(stderr, "Unable to find usb device.");
+		goto enum_out;
+	}
 
```

Comment:
I think this check is not necessary, as path exists guaranteed the device node should be there

Reply:
The [documentation](https://man7.org/linux/man-pages/man3/udev_device_new_from_syspath.3.html) says:

> On success, udev_device_new_from_syspath() ... return a pointer to the allocated udev device. On failure, NULL is returned, and errno is set appropriately.

If it's allocating it can fail, so I just wanted to be safe.